### PR TITLE
Minor: reduce code duplication in PruningPredicate test

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/arrow-datafusion/pull/8440

## Rationale for this change
I made a function that was more concise and easier to understand than  current tests which have substantial duplication, as part of https://github.com/apache/arrow-datafusion/pull/8440, but wanted to keep that PR as small as possible to make review easier. 

## What changes are included in this PR?
1. refactor other tests to use `prune_with_expr` to reduce code duplication

## Are these changes tested?
all tests

### CODE REVIEW

File: datafusion/core/src/physical_optimizer/pruning.rs [hunk 1]
```diff
@@ -2004,100 +2004,97 @@ mod tests {
...
-        let expr = logical2physical(&expr, &schema);
-        // If the data is written by spark, the physical data type is INT32 in the parquet
-        // So we use the INT32 type of statistic.
-        let statistics = TestStatistics::new().with(
-            "s1",
-            ContainerStats::new_i32(
-                vec![Some(0), Some(4), None, Some(3)], // min
-                vec![Some(5), Some(6), Some(4), None], // max
+
+        prune_with_expr(
```
Comment (alamb) [inline datafusion/core/src/physical_optimizer/pruning.rs:2008]: The idea is to reduce the boiler plate required to setup the test / prune the expression, and simply list the data and expected results

**Comment by alamb** [comment]
cc @yahoNanJing 

@crepererum do you have time to review this PR?

**Comment by alamb** [comment]
@Ted-Jiang  I wonder if you have some time to review this PR?

**Review by viirya — APPROVED** [review]
I think the most valuable about `prune_with_expr` is it makes the test more readable.

**Review by Ted-Jiang — APPROVED** [review]
👍

**Comment by alamb** [comment]
Thank you @viirya  and @Ted-Jiang

**Pull request merged**